OrbitCore/LinuxPerfEventProcessor.cpp

```diff
@@ -39,8 +41,12 @@ void LinuxPerfEventProcessor::ProcessAll()
 void LinuxPerfEventProcessor::ProcessTillOffset()
 {
     // Process the events in the event_queue
+    // We do not lock this read, as otherwise, 
+    //  it would lock the complete exceution of 
+    //  that function.
     while(!m_EventQueue.empty())
     {
+        ScopeLock lock(m_Mutex);
```

Comment:
Does event->accept() need to be protected by the lock?  To minimize contention, could we restrict the lock to the dequeue operation?  Maybe implement a TryDequeue() method that would return nullptr if there is nothing in the queue?